crates/viewer/re_space_view_spatial/src/video_cache.rs

```diff
@@ -0,0 +1,79 @@
+use egui::mutex::Mutex;
+use re_entity_db::VersionedInstancePathHash;
+use re_log_types::hash::Hash64;
+use re_renderer::renderer::Video;
+use re_renderer::RenderContext;
+use re_types::components::MediaType;
+use re_viewer_context::Cache;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
+// ----------------------------------------------------------------------------
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct VideoCacheKey {
+    pub versioned_instance_path_hash: VersionedInstancePathHash,
+    pub query_result_hash: Hash64,
+    pub media_type: Option<MediaType>,
+}
+
+struct Entry {
+    used_this_frame: AtomicBool,
+    // TODO(jan): warn_once instead of having option here
+    //            who cares if it wastes cycles loading the video
```

Comment:
I think the `Option` is the right approach; otherwise `Video::load` will be called every frame on failure, which could be a huge CPU drain